### ISSUE TITLE
Octopus Intelligent planned dispatches not updating

### ISSUE THREAD

**Issue by chris-y** [BottlecapDave/HomeAssistant-OctopusEnergy#1088]
### Describe the bug

The Planned Dispatches is not showing spots that get added later.
I had a planned charge between 0400-0500 which for some reason extended up to 0830. This is showing in Completed Dispatches, but did not appear in Planned Dispatches (at all as far as I can tell - at least the integration never picked it up as I have a notification set for when it changes). The binary sensor also didn't show that there was a dispatch in progress throughout this time.

Completed dispatches: 

![Screenshot_20241120-175137](https://github.com/user-attachments/assets/0b1d82b3-60f0-4778-ab6b-e748f80acb68)

When binary sensor was active:
![Screenshot_20241120-191712](https://github.com/user-attachments/assets/9c3236ee-2d07-42f3-a533-b2da56fd7c72)

### Reproduction steps

Not sure! I use the Renault IOG integration, it could be a bug with that (it is beta), or it could be that this integration didn't pick the new times up.

### Expected behaviour

I'd expect the binary sensor to switch on, even if these are last minute additions to the schedule.

### Tariff Code

E-1R-INTELLI-VAR-22-10-14-A

### Integration Version

13.1.3

### Home Assistant Version

2024.11.2

### Fresh Install?

Not specified

### Home Assistant Logs

I've started debug logging in hope that it happens again tonight.

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by chris-y** [comment]
Related: I plugged the car in and it started smart charging and allocated some slots. I then disabled smart charging in the Mobilize app - this doesn't appear to link to the smart charging option in the Octopus API. It also left my slots in the upcoming dispatches.

I hope this will be sorted out by Octopus, even if it's a one way connection. I don't think there's anything this integration can do to work around it at the moment.

**Comment by BottlecapDave** [comment]
Hello and sorry you saw this issue. Did you have any errors in your logs? There are sometimes instances where OE doesn't update the planned dispatches, which is what the integration uses to power the dispatching sensor, which might be what happened in your scenario.

For your second situation, if there is no link between mobilize and Octopus then this integration won't know either. It just recreates what's available in the app.

**Comment by BottlecapDave** [comment]
If you're comfortable with running SQL queries, you can run the following via the SQL add-on to see what the attributes were at the time of the dispatching sensor turning on/off

```
SELECT  "states_meta".entity_id, "states".state,  "state_attributes".shared_attrs, DATETIME("states".last_updated_ts, 'unixepoch') AS last_updated
FROM "states_meta"
JOIN "states" ON "states".metadata_id = "states_meta".metadata_id
JOIN "state_attributes" ON "state_attributes".attributes_id = "states".attributes_id 
WHERE "states_meta".entity_id LIKE '%intelligent_dispatching%'
AND DATETIME("states".last_updated_ts, 'unixepoch') >= '2024-11-20 00:00:00Z'
AND DATETIME("states".last_updated_ts, 'unixepoch') <= '2024-11-20 10:00:00Z'
```

It would also be good to upvote https://community.home-assistant.io/t/simple-way-to-view-the-history-of-sensor-attributes/553324 to make this process easier

**Comment by chris-y** [comment]
![Screenshot_20241124-111127-091.png](https://github.com/user-attachments/assets/da8ed448-950a-4802-bc9f-cef724115b7e)

**Comment by chris-y** [comment]
Yeah, I'm suspecting that the planned dispatches never got updated. They were showing as completed after they had passed.

**Comment by BottlecapDave** [comment]
The screenshot isn't very helpful (as I can't see the planned dispatches at 4am), but yeah it looks like there were no planned dispatches at 5:30am which is why it turned off. I take the data as is from OE API, so in this scenario the integration worked as intended with the data it had.

Because of this, I'll close this issue, but if you see it again with the correct data being present, please open a new issue.
